contracts/adapters/AutoRepayAndBorrowAdapter.sol

```diff
@@ -88,16 +102,59 @@ contract AutoRepay is BaseUniswapAdapter {
     return addresses;
   }
 
-  function setMinMaxHealthFactor(uint256 minHealthFactor, uint256 maxHealthFactor) external {
+  function setMinTargetHealthFactor(uint256 minHealthFactor, uint256 targetHealthFactor) external {
+    require(
+      targetHealthFactor >= minHealthFactor,
+      'targetHealthFactor should be more or equal than minHealthFactor'
+    );
+    userInfos[msg.sender] = UserInfo({
+      minHealthFactor: minHealthFactor,
+      targetHealthFactor: targetHealthFactor,
+      maxHealthFactor: 0,
+      rateMode: 0,
+      borrowAddress: address(0),
+      collateralAddress: address(0)
+    });
+    emit HealthFactorSet(msg.sender, minHealthFactor, targetHealthFactor, 0, 0, address(0), address(0));
+  }
+
+  function setMinTargetMaxHealthFactor(uint256 minHealthFactor, uint256 targetHealthFactor, uint256 maxHealthFactor, address borrowAddress, address collateralAddress, uint256 rateMode) external {
+    require(
+      targetHealthFactor >= minHealthFactor,
+      'TargetHealthFactor should be more or equal than minHealthFactor'
+    );
     require(
-      maxHealthFactor >= minHealthFactor,
-      'maxHealthFactor should be more or equal than minHealthFactor'
+      maxHealthFactor >= targetHealthFactor,
+      'MaxHealthFactor should be more or equal than targetHealthFactor'
     );
+    // 1 for Stable, 2 for Variable
+    require(rateMode == 1 || rateMode == 2, 'Not valid rate mode provided');
+    require(_getReserveData(collateralAddress).aTokenAddress != address(0), 'Not valid collateralAddress provided');
+    require(_getReserveData(borrowAddress).aTokenAddress != address(0), 'Not valid borrowAddress provided');
+    require(collateralAddress != borrowAddress, 'Collateral and borrow could not be equal');
+
     userInfos[msg.sender] = UserInfo({
       minHealthFactor: minHealthFactor,
-      maxHealthFactor: maxHealthFactor
+      targetHealthFactor: targetHealthFactor,
+      maxHealthFactor: maxHealthFactor,
+      rateMode: rateMode,
+      collateralAddress: collateralAddress,
+      borrowAddress: borrowAddress
     });
-    emit HealthFactorSet(msg.sender, minHealthFactor, maxHealthFactor);
+    emit HealthFactorSet(msg.sender, minHealthFactor, targetHealthFactor, maxHealthFactor, rateMode, borrowAddress, collateralAddress);
+  }
+
+  function clearMinTargetMaxHealthFactor() external {
+    require(userInfos[msg.sender].minHealthFactor > 0, 'Already clear');
+    userInfos[msg.sender] = UserInfo({
+      minHealthFactor: 0,
+      targetHealthFactor: 0,
+      maxHealthFactor: 0,
+      rateMode: 0,
+      collateralAddress: address(0),
+      borrowAddress: address(0)
+    });
+    emit HealthFactorSet(msg.sender, 0, 0, 0, 0, address(0), address(0));
   }
 
   function _checkMinHealthFactor(address user) internal view returns (uint256) {
```

Comment:
Would be more complicated if   _checkMinHealthFactor(address user) and _checkMaxHealthFactor(address user) are uinted into the single function something like _checkHealthFactor(address user, bool max), because they both have quite simular functionality and the only difference in condition.
healthFactor **<** userInfos[user]**.minHealthFactor,**
healthFactor **>** userInfos[user]**.maxHealthFactor,**